asf_heat_pump_suitability/analysis/exploratory/subset_epc_north_england.py

```diff
@@ -0,0 +1,55 @@
+"""
+Subset the EPC data to only have data from local authorities which are above Liverpool (and including Liverpool)
+"""
+
+import polars as pl
+from asf_heat_pump_suitability import config
+from asf_heat_pump_suitability.pipeline.prepare_features import lat_lon, output_areas
+
+import s3fs
+
+if __name__ == "__main__":
+    epc_path = (
+        "s3://asf-daps/lakehouse/processed/epc/deduplicated/processed_dedupl-0.parquet"
+    )
+    epc_df = pl.read_parquet(epc_path, columns=config["usecols"]["epc"])
+
+    # Add lat/long and local authority per UPRN
+    uprn_latlon_df = lat_lon.transform_df_osopen_uprn_latlon()
+    epc_df = epc_df.join(uprn_latlon_df, how="left", on="UPRN")
+
+    epc_df = output_areas.standardise_col_postcode(epc_df, pcd_col="POSTCODE")
+    onspd_df = output_areas.transform_df_ons_pd()
+    epc_df = epc_df.join(onspd_df, how="left", on="POSTCODE")
+
+    # Replace `lad_code` from postcode with `lad_code` from geospatial join and postcode
+    uprn_lad_df = output_areas.sjoin_df_uprn_lad_code(epc_df)  # Takes a long time!
+    epc_df = epc_df.drop("lad_code").join(uprn_lad_df, how="left", on="UPRN")
+
+    # Get the UPRNs which are from LAs North of Liverpool
+    # Want to include the entire LA of data (so can't just subset on latitude alone)
+
+    # Find minimum latitude per LAD
+    min_lat_per_lad = epc_df[["lad_code", "LATITUDE"]].group_by("lad_code").min()
```

Comment:
I think this will give you the latitude of the southern-most UPRN in the local authority, not necessarily the southern-most tip of the LA.

Since this script is for sampling purposes I don't think it matters but might be worth documenting into the code.

Reply:
oh yeh that's true! will document to make that clear

Reply:
good spot